hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactionPipeline.java

```diff
@@ -223,10 +223,16 @@ public boolean flattenOneSegment(long requesterVersion,
         LOG.warn("Segment flattening failed, because versions do not match");
         return false;
       }
-      int i = 0;
+      int i = -1;
       for (ImmutableSegment s : pipeline) {
+        i++;
```

Comment:
What is the reason for this change?

Reply:
@Apache9， because  In my patch，I move the `if (!currActive.isEmpty())` check from the `CompactingMemStore.pushActiveToPipeline` in write path  to the background `InMemoryCompactionRunnable` after `currActive.waitForUpdates` is called, so here `s` may be empty . If `s` is empty , the flatten should be skipped , and because the i++ is at the end of the loop, so I move the i++ to the start of the loop and initialized it from 0 to `-1`

Reply:
OK. Got it.